uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/test/TestRunEventCollection.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.guvnor.common.services.backend.test;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.event.Event;
+
+import org.guvnor.common.services.shared.test.Failure;
+import org.guvnor.common.services.shared.test.TestResultMessage;
+
+public class TestRunEventCollection
+        implements Event<TestResultMessage> {
```

Comment:
To be honest I don't like to implement an J2EE annotation as "fake" class to aggregate the result. I understood you did this to reuse the same `runTest` method but I think it is more understandable to create a new wrapper/interface like `TestResultConsumer` that accept `TestResultMessage`. Then create two implementation of that consumer, one `AccumulatorConsumer` and one `CdiEventConsumer` that do the proper firing.
Wdyt? It is also fine to keep it as is but add at least some javadoc to this class

Reply:
Yeah I moved away from this with the recent update. The events are no longer used to report a single test run for example, so no need to abuse the Event.